README.md

```diff
@@ -325,6 +325,50 @@ sequencer.insertSteps({
 return value: **`sequencer`** (To allow method chaining)
 
 
+## Creating a User Interface
+
+Image Sequencer provides the following events which can be used to generate a UI:
+
+* `onSetup` : this event is triggered when a new module is set up. This can be used,
+for instance, to generate a DIV element to store the generated image for that step.
+* `onDraw` : This event is triggered when Image Sequencer starts drawing the output
+for a module. This can be used, for instance, to overlay a loading GIF over the DIV
+generated above.
+* `onComplete` : This event is triggered when Image Sequencer has drawn the output
+for a module. This can be used, for instance, to update the DIV with the new image
+and remove the loading GIF generated above.
+* `onRemove` : This event is triggered when a module is removed. This can be used,
+for instance, to remove the DIV generated above.
+
+How to define these functions:
+
+```js
+sequencer.setUI({
+  onSetup: function() {},
```

Comment:
Yes I think it already is what you are talking about. Generic functions can be written.
The user has access to :
* What image the step is applied to
* Details of the step
* Details of the client : browser/Node

This is all the data which is needed to make a UI, I feel.

Although we could give access to the original image, and the metadata, but I am not really clear about what you are saying there...

I don't think we should ask to return generated HTML. That is because, we don't really need it. The user has all access to the front-end of the code. We are providing all the data the user might need and he can implement it to the front-end as he wishes.